paasta_tools/kubernetes_tools.py

```diff
@@ -1855,20 +1853,26 @@ def sanitize_for_config_hash(
         :param config: complete_config hash to sanitise
         :returns: sanitised copy of complete_config hash
         """
-        ahash = {
-            key: copy.deepcopy(value)
-            for key, value in config.to_dict().items()
-            if key not in CONFIG_HASH_BLACKLIST
-        }
-        spec = ahash["spec"]
-        ahash["spec"] = {
-            key: copy.deepcopy(value)
-            for key, value in spec.items()
-            if key not in CONFIG_HASH_BLACKLIST
-        }
+        ahash = config.to_dict()  # deep convert to dict
         ahash["paasta_secrets"] = get_kubernetes_secret_hashes(
             service=self.get_service(), environment_variables=self.get_env()
         )
+
+        # remove data we dont want used to hash configs
+        # replica count
+        if ahash.get("spec", {}).get("replicas"):
```

Comment:
should this have an `is None` tacked on the end? we wouldn’t delete this key if there are 0 replicas otherwise (unless we never enter this function in that case)